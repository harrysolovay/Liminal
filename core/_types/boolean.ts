import type { Type } from "../Type.ts"
import { declareType } from "./_declareType.ts"

export const boolean: Type<boolean> = declareType(() => boolean)
