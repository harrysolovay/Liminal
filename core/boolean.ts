import type { Type } from "../Type.ts"
import { declare } from "../TypeDeclaration.ts"

export const boolean: Type.Initial<boolean> = declare({
  name: "boolean",
  source: {
    getType: () => boolean,
  },
  subschema: () => ({ type: "boolean" }),
})
