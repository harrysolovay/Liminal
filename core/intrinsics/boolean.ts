import { declareType } from "../declareType.ts"
import type { Type } from "../Type.ts"

export const boolean: Type<boolean> = declareType({
  type: "boolean",
  self() {
    return boolean
  },
})
