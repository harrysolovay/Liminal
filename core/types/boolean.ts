import { Type } from "../Type.ts"

export const boolean: Type<boolean> = Type({
  name: "boolean",
  source: {
    getType: () => boolean,
  },
})
