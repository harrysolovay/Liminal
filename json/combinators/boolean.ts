import { Type } from "../../core/mod.ts"

export const boolean: Type<boolean> = Type({
  getType: () => boolean,
})
