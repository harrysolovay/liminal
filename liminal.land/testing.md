# Liminal Testing <Badge type="warning" text="beta" />
