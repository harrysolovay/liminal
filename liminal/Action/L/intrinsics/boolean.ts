import { declare } from "../declare.ts"
import type { L } from "../L.ts"

export const boolean: L<boolean> = declare(() => boolean)
