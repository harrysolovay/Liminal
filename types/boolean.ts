import { declare, type Type } from "../core/mod.ts"

export const boolean: Type.Initial<boolean> = declare({
  name: "boolean",
  source: {
    getType: () => boolean,
  },
  subschema: () => ({ type: "boolean" }),
})
