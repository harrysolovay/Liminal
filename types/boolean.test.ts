import { assertTypeSnapshot } from "testing"
import { boolean } from "./boolean.ts"

Deno.test("boolean", async (t) => {
  await assertTypeSnapshot(t, boolean)
})
