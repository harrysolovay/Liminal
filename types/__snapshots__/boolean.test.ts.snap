export const snapshot = {};

snapshot[`boolean 1`] = `
{
  type: "boolean",
}
`;

snapshot[`boolean 2`] = `
{
  description: "One.",
  type: "boolean",
}
`;

snapshot[`boolean 3`] = `
{
  description: "One. Two.",
  type: "boolean",
}
`;
